src/config/__tests__/settings.spec.ts

```diff
@@ -0,0 +1,228 @@
+import fs from 'fs';
+
+import * as settings from '../settings';
+import { PathManagementStrategy } from '@/integrations/pathManager';
+import { RecursivePartial } from '~/utils/typeUtils';
+
+describe('updateFromCommandLine', () => {
+  let prefs: settings.Settings;
+  let configDir: string;
+
+  beforeEach(() => {
+    jest.spyOn(fs, 'writeFileSync').mockImplementation(() => { });
+    prefs = {
+      version:    4,
+      kubernetes: {
+        version:                    '1.23.5',
+        memoryInGB:                 4,
+        numberCPUs:                 2,
+        port:                       6443,
+        containerEngine:            settings.ContainerEngine.MOBY,
+        checkForExistingKimBuilder: false,
+        enabled:                    true,
+        WSLIntegrations:            {},
+        options:                    {
+          traefik: true,
+          flannel: false,
+        },
+        suppressSudo: false
+      },
+      portForwarding: { includeKubernetesServices: false },
+      images:         {
+        showAll:   true,
+        namespace: 'k8s.io'
+      },
+      telemetry:              true,
+      updater:                true,
+      debug:                  true,
+      pathManagementStrategy: PathManagementStrategy.NotSet,
+    };
+  });
+
+  describe('getUpdatableNode', () => {
+    test('returns nil on an invalid top level accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'blah-blah-blah');
+
+      expect(result).toBeNull();
+    });
+    test('returns nil on an invalid internal accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'kubernetes-options-blah');
+
+      expect(result).toBeNull();
+    });
+    test('returns the full pref with a top-level accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'kubernetes');
+
+      if (!result) {
+        expect(result).not.toBeNull();
+      } else {
+        const lhs: RecursivePartial<settings.Settings> = result[0];
+        const accessor: string = result[1];
+
+        expect(lhs).toMatchObject(prefs);
+        expect(accessor).toBe('kubernetes');
+      }
```

Comment:
Perhaps we can just force the return type:
```suggestion
      const result = settings.getUpdatableNode(prefs, 'kubernetes') as [Record<string, any>, string];

      expect(result).not.toBeNull();
      const [lhs, accessor] = result;

      expect(lhs).toMatchObject(prefs);
      expect(accessor).toBe('kubernetes');
```

Reply:
Yes, no need for the recursive type